docs/source/developer/committer-notes.rst

```diff
@@ -0,0 +1,162 @@
+
+* For major and minor releases, create a new release branch. Release branches have the naming scheme ``release/<MAJ>.<MIN>.x`` (where ``x`` is literal "``x``"). For example, version ``1.2.0`` → branch ``release/1.2.x``. Release branches remain alive as long as that minor version is supported. (Note: Release branch names *don't* contain the ``-rc.<NUM>`` suffix, even when doing an RC release. All RC releases and the GM release use the same branch.)
+* For patch releases, don’t create a new release branch, use the existing branch for the corresponding major or minor version. For example, version = ``1.2.1`` → branch ``release/1.2.x``.
+* For all releases, create a tag. Tags have the naming scheme ``v<MAJ>.<MIN>.<PAT>[-rc.<NUM>]``. For example, for version = ``1.2.0``, tag = ``v1.2.0``. (Note: Tag names *do* contain the ``-rc.<NUM>`` suffix, when doing an RC release.)
+* Use RC releases to validate the system end-to-end before creating the GM release. There should be at least one successful RC release before creating the GM release.
+* Use a GitHub Release to initiate each OpenDP release. This will run the GitHub Workflows that handle the build and publish process (see below).
+
+Playbook
+--------
+
+#. Identify names:
+
+   ==============  ===============================  =================  ==============================
+   Item            Format                           Example            Notes
+   ==============  ===============================  =================  ==============================
+   base version    ``<MAJ>.<MIN>.<PAT>``            ``1.2.0``          base for RC and GM versions
+   release branch  ``release/<MAJ>.<MIN>.x``        ``release/1.2.x``  branch used for all iterations
+   RC version      ``<MAJ>.<MIN>.<PAT>-rc.<NUM>``   ``1.2.0-rc.1``     incremented for each RC
+   RC tag          ``v<MAJ>.<MIN>.<PAT>-rc.<NUM>``  ``v1.2.0-rc.1``
+   GM version      ``<MAJ>.<MIN>.<PAT>``            ``1.2.0``          same as base version
+   GM tag          ``v<MAJ>.<MIN>.<PAT>``           ``v1.2.0``
+   ==============  ===============================  =================  ==============================
+
+#. Create/update the release branch:
+
+   * Major or minor release ONLY: Create a *new* release branch, based on the desired point in ``main``.
+   * Patch release ONLY: Use the *existing* branch from the previous major or minor release, and cherry-pick changes from ``main`` into the release branch.
+
+#. Set the RC number to 1.
+#. Specify the version for this iteration: ``<MAJ>.<MIN>.<PAT>[-rc.<NUM>]``
+#. Update the version field(s) in the following files:
+
+   * ``VERSION``
+   * ``rust/opendp/Cargo.toml``
+   * ``rust/opendp-ffi/Cargo.toml`` (two entries!!!)
+   * ``python/setup.cfg`` (TODO: automate this from sphinx conf.py?)
```

Comment:
I do think we should include Sphinx in some way, even if it's just a separate bullet for now.

Reply:
Yeah for now it's easier to update all files manually, I'll mention the sphinx config too.